src/transformers/models/wav2vec2/modeling_flax_wav2vec2.py

```diff
@@ -766,6 +766,75 @@ def __call__(self, hidden_states, mask_time_indices=None, deterministic=True, te
         return codevectors, perplexity
 
 
+class FlaxWav2Vec2Adapter(nn.Module):
+    config: Wav2Vec2Config
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        # hidden_states require down-projection if feature dims don't match
+        if self.config.output_hidden_size != self.config.hidden_size:
+            self.proj = nn.Dense(
+                self.config.output_hidden_size,
+                kernel_init=jax.nn.initializers.normal(self.config.initializer_range),
+                dtype=self.dtype,
+            )
+            self.proj_layer_norm = nn.LayerNorm(epsilon=self.config.layer_norm_eps, dtype=self.dtype)
+        else:
+            self.proj = self.proj_layer_norm = None
+
+        self.layers = FlaxWav2Vec2AdapterLayersCollection(self.config, dtype=self.dtype)
+
+    def __call__(self, hidden_states, deterministic=True):
+        # down-project hidden_states if required
+        if self.proj is not None and self.proj_layer_norm is not None:
+            hidden_states = self.proj(hidden_states)
+            hidden_states = self.proj_layer_norm(hidden_states)
+
+        hidden_states = self.layers(hidden_states)
+
+        return hidden_states
+
+
+class FlaxWav2Vec2AdapterLayer(nn.Module):
+    config: Wav2Vec2Config
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.conv = nn.Conv(
+            features=2 * self.config.output_hidden_size,
+            kernel_size=(self.config.adapter_kernel_size,),
+            strides=(self.config.adapter_stride,),
+            padding=((1, 1),),
+            kernel_init=jax.nn.initializers.normal(self.config.initializer_range),
+            dtype=self.dtype,
+        )
+
+    def __call__(self, hidden_states):
+        hidden_states = self.conv(hidden_states)
+        hidden_states = nn.glu(hidden_states, axis=2)
+
+        return hidden_states
+
+
+class FlaxWav2Vec2AdapterLayersCollection(nn.Module):
+    config: Wav2Vec2Config
+    dtype: jnp.dtype = jnp.float32
+
+    def setup(self):
+        self.layers = [
+            FlaxWav2Vec2AdapterLayer(self.config, name=str(i), dtype=self.dtype)
+            for i in range(self.config.num_adapter_layers)
+        ]
+
+    def __call__(self, hidden_states, deterministic=True):
+        for conv_layer in self.layers:
+            layerdrop_prob = np.random.random()
```

Comment:
```suggestion
```
One should more or less never use `np.random.random()` in a Flax model because it makes the model non-deterministic. Instead it's better to force the model to pass a PRNGKey at `.apply(...)` (shown here: https://github.com/huggingface/transformers/blob/8ce133063120683018b214fe10d1449e4c2401da/src/transformers/models/wav2vec2/modeling_flax_wav2vec2.py#L1197) and then retrieve it with `self.make_rng(...)` as shown here: https://github.com/huggingface/transformers/blob/8ce133063120683018b214fe10d1449e4c2401da/src/transformers/models/wav2vec2/modeling_flax_wav2vec2.py#L743 . Think `layerdrop_prob` is really neat but so far we have just left it out in Flax. So maybe we can just remove it here for now as well and only add it in the "correct" way in a follow-up PR (not super important though)